src/Lexer.php

```diff
@@ -42,20 +55,26 @@ class Lexer
     private int $lineNumber;
     private bool $eos;
     /**
-     * @var array<string, string>
+     * @phpstan-var array<TKeywordsType, TKeywordsString>
      */
     private array $keywordsCache = [];
     /**
-     * @var array<string, list<string>>
+     * @var array{}|array{
+     *     Given: list<string>,
+     *     When: list<string>,
+     *     Then: list<string>,
+     *     And: list<string>,
+     *     But: list<string>,
+     * }
      */
     private array $stepKeywordTypesCache = [];
     /**
-     * @phpstan-var list<TToken>
+     * @phpstan-var list<TNullValueToken|TStringValueToken>
```

Comment:
Sorry, I'm really not seeing the benefit of these changes. The deferred / stashed tokens can by definition be any type of token, with any of the possible additional array keys. `TToken` makes clear that we don't know what type of token we have (indeed deferred tokens are passed through the public API and could in theory be anything). `TNullValueToken|TStringValueToken` implies that it is somehow a subset?

I'd understand more if these were real classes where e.g. `TStepToken` extends `TStringValueToken` extends `TToken` - but they don't. IMO this is adding a lot of complexity (and changes and code reviews) as well as potentially false confidence, and I'm not sure what we gain in return?

Reply:
Define "these changes".. in some cases, such as this one, I was forced to change it because I had to split `TToken`.

Now that I think about it, maybe I should define `TToken` as `TNullValueToken|TStringValueToken|etc`?

Reply:
I went back to that union approach, it should be cleaner and less wrong. :)